crates/store/re_log_encoding/src/codec/file/mod.rs

```diff
@@ -30,6 +30,15 @@ pub(crate) struct MessageHeader {
 }
 
 impl MessageHeader {
+    /// Size of an encoded message header, in bytes.
+    // NOTE: This is `size_of` on a `repr(Rust)` struct,
+    //       which is fine because we control the layout
+    //       in the definition above, and tests would quickly
+    //       catch any sort of regression.
```

Comment:
Why are we not switching to the C ABI if we need control over layout though?

Reply:
I thought about this a bit more, and maybe this should just be a literal constant. We need control over the _encoding_, not the layout of the struct